inbound.go

```diff
@@ -267,6 +270,12 @@ func (call *InboundCall) RemotePeer() PeerInfo {
 	return call.conn.RemotePeerInfo()
 }
 
+// TLSConnectionState returns the TLS connection state of the call when the
+// underlying connection is TLS.
+func (call *InboundCall) TLSConnectionState() *tls.ConnectionState {
+	return call.tlsConnectionState
+}
+
```

Comment:
Instead of tying tchannel to TLS explicitly (which seems like an unnecessary dependency), It seems better to expose just the underlying `net.Conn`, and outside of TChannel, callers can try to cast the `net.Conn` to a `*tls.Conn` and extract the connection state.

Reply:
Exposing `net.Conn` seems like a good idea, though exposing it outside could be wrongly used by the callers? (primarily YARPC, so we should be good).

Reply:
Agree, it could be misused, though it should be possible to do something similar today: wrapping the listener + dialer allows callers to get access to connections and misuse them (just a little more obscure). Since the `InboundCall` isn't accessible to standard clients anyway, it'll still remain relatively obscure!